tests/cockpit-tasks-restricted.json

```diff
@@ -0,0 +1,146 @@
+{
```

Comment:
For easier review, this is the [diff compared to tests/cockpit-tasks.json](https://paste.fedoraproject.org/paste/6l2fBfSuMXfEFScFGTaS4g). If we can make this service work without the `uid == 1111` assumption, is there any reason to not do this on our current Cockpit openshift cluster as well? I. e. also apply the `securityContext` and `hack-hide-virtual-net` changes to tests/cockpit-tasks.json?

AFAICS the only significant change is about not mounting the host's tests/cockpit-tasks.json, but using a regular docker volume for that as well. Likewise, this would also simplify things on our current OpenShift cluster, as we have to jump through quite a lot of hoops to make this work (remember to do the chown, apply the appropriate SELinux context, etc.). It seems to me that we should just use regular volumes there as well.

So in summary: I don't see anything in that diff that shouldn't apply to our current cluster as well - i. e. WDYT about patching tests/cockpit-tasks.json  instead and we roll this out on our current cluster? (I'm fine with hand-holding this and reverting if necessary).

Reply:
It's just *way* less efficient. We can run the restricted one in our current cluster, but we could never achieve the density we do today.

Reply:
So I'm obviously missing something here. Docker volumes get shared between all containers that use them. I. e. they have much the same behaviour as with our current host bind mount, except that the cached data isn't (easily) visible on the host (but that's not very important either). What is the principal difference in density here? Of course for this it should use an actual volume, not an `emptyDir`.

My understanding is that [OpenShift volumes](https://docs.openshift.com/container-platform/3.7/dev_guide/volumes.html) are pretty much exactly docker volumes.

Or is the thing that I'm missing that centos CI does not have volumes? I can't believe that, as (1) you are already using a secrets volume, and (2) such an OpenShift environment would be almost useless. I'm certainly not willing to move off to such a cluster with non-shared caches between pods, that would be insanely inefficient (not just for storage, also for test runtime).

Reply:
CentOS Ci has volumes, just no /dev/kvm support (like oci-kvm-hook yet).

Openshift volumes have semantics that require strange fsGroup or a number of optional techniques in order to get containers to write to them. 

Openshift went and made docker containers run under a random UID, which has *all* sorts of strange implications. In this case we don't need volumes, since we're just using this stuff as a cache. Booting a VM image from a Gluster or NFS volume is a fail in any case. We really just want the container to pull images to an emptyDir and use that.

Reply:
I don't understand - why wouldn't we want to use a shared volume? Your intent with the `chmod a+w` is obviously to make the cache world-writable (except that happens in the wrong place - it needs to happen when the cache dirs are written, not in the static container image). With random UIDs you can indeed not do any better than making volumes world-writable (but also, it's fairly irrelevant in this environment).